replication/binlogsyncer.go

```diff
@@ -128,6 +128,12 @@ type BinlogSyncerConfig struct {
 
 	EventCacheCount int
 
+	// MariaDBDynamicLogPos enables dynamic LogPos calculation for MariaDB.
+	// When enabled, automatically adds BINLOG_SEND_ANNOTATE_ROWS_EVENT flag
+	// to ensure correct position calculation in MariaDB 11.4+.
+	// Only works with MariaDB flavor.
+	MariaDBDynamicLogPos bool
```

Comment:
in `NewBinlogSyncer` we can check the flavor and MariaDBDynamicLogPos match, and print warning log if mismatch.